### PR TITLE
[MM-55746] Add documentation on how to use new `ServeMetrics` plugin hook

### PR DESCRIPTION
**Pull request by streamer45**
#### Summary

Adding some documentation under plugins development "Best practices" section on how to use the newly introduced [`ServeMetrics`](https://github.com/mattermost/mattermost/pull/24249) plugin hook.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-55746

### CODE REVIEW

**Comment by streamer45** [comment]
Preview of changes can be found at http://mattermost-dev-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/1325/integrate/plugins/best-practices/#how-to-expose-performance-metrics-for-a-plugin

File: site/content/integrate/plugins/best-practices.md [hunk 1]
```diff
@@ -128,3 +128,34 @@ GetUsers(options *model.UserGetOptions) ([]*model.User, *model.AppError)
 ```
 
 Old servers won't do anything with new, unrecognized fields, but also won't break if they are present.
+
+
+## How to expose performance metrics for a plugin?
+
+Starting in Mattermost v9.4 we provide a new [`ServeMetrics`]({{< ref "/integrate/reference/server/server-reference#API.ServeMetrics" >}}) hook that can be used to expose performance metrics in the [open metrics format](https://openmetrics.io/) under the common HTTP listener controlled by the [`MetricsSettings.ListenAddress`](https://docs.mattermost.com/configure/environment-configuration-settings.html#listen-address-for-performance) config setting.
```
Comment (cwarnermm) [inline site/content/integrate/plugins/best-practices.md:135]:
```suggestion
From Mattermost v9.4, a [`ServeMetrics`]({{< ref "/integrate/reference/server/server-reference#API.ServeMetrics" >}}) hook can be used to expose performance metrics in the [open metrics format](https://openmetrics.io/) under the common HTTP listener controlled by the [`MetricsSettings.ListenAddress`](https://docs.mattermost.com/configure/environment-configuration-settings.html#listen-address-for-performance) config setting.
```

**Review by cwarnermm — APPROVED** [review]
Thanks, @streamer45!!

**Review by lieut-data — APPROVED** [review]

**Comment by streamer45** [comment]
@cwarnermm It should go with the upcoming 9.4 release.

**Pull request merged**